src/main/java/org/retest/rebazer/connector/GithubConnector.java

```diff
@@ -0,0 +1,136 @@
+
+public class GithubConnector implements RepositoryConnector {
+
+	private final static String baseUrl = "https://api.github.com/";
+
+	private final RestTemplate template;
+
+	public GithubConnector( final RepositoryTeam repoTeam, final RepositoryConfig repoConfig,
+			final RestTemplateBuilder builder ) {
+		final String basePath = "/repos/" + repoTeam.getName() + "/" + repoConfig.getName();
+
+		template = builder.basicAuthorization( repoTeam.getUser(), repoTeam.getPass() ).rootUri( baseUrl + basePath )
+				.build();
+	}
+
+	@Override
+	public PullRequest getLatestUpdate( final PullRequest pullRequest ) {
+		final DocumentContext jsonPath = jsonPathForPath( requestPath( pullRequest ) );
+		final PullRequest updatedPullRequest = PullRequest.builder() //
+				.id( pullRequest.getId() ) //
+				.source( pullRequest.getSource() ) //
+				.destination( pullRequest.getDestination() ) //
+				.lastUpdate( jsonPath.read( "$.updated_at" ) ) //
+				.build();
+		return updatedPullRequest;
+	}
+
+	@Override
+	public boolean isApproved( final PullRequest pullRequest ) {
+		final DocumentContext jsonPath = jsonPathForPath( requestPath( pullRequest ) + "/reviews" );
+		final List<String> states = jsonPath.read( "$..state" );
+		boolean approved = false;
+		for ( final String state : states ) {
+			if ( state.equals( "APPROVED" ) ) {
+				approved = true;
+			} else {
+				approved = false;
+			}
+		}
+		return approved;
+	}
+
+	@Override
+	public boolean rebaseNeeded( final PullRequest pullRequest ) {
+		return !getLastCommonCommitId( pullRequest ).equals( getHeadOfBranch( pullRequest ) );
+	}
+
+	String getHeadOfBranch( final PullRequest pullRequest ) {
+		return jsonPathForPath( "/git/refs/heads/" + pullRequest.getDestination() ).read( "$.object.sha" );
+	}
+
+	String getLastCommonCommitId( final PullRequest pullRequest ) {
+		final DocumentContext jsonPath = jsonPathForPath( requestPath( pullRequest ) + "/commits" );
+
+		final List<String> commitIds = jsonPath.read( "$..sha" );
+		final List<String> parentIds = jsonPath.read( "$..parents..sha" );
+
+		return parentIds.stream().filter( parent -> commitIds.contains( parent ) ).findFirst()
+				.orElseThrow( IllegalStateException::new );
+	}
+
+	@Override
+	public void merge( final PullRequest pullRequest ) {
+		final String message = String.format( "Merged in %s (pull request #%d) by ReBaZer", pullRequest.getSource(),
+				pullRequest.getId() );
+		final Map<String, String> request = new HashMap<>();
+		request.put( "commit_title", message );
+		request.put( "merge_method", "merge" );
+
+		template.put( requestPath( pullRequest ) + "/merge", request, Object.class );
+	}
+
+	@Override
+	public boolean greenBuildExists( final PullRequest pullRequest ) {
+		final String urlPath = "/commits/" + pullRequest.getSource() + "/status";
+		final DocumentContext jsonPath = jsonPathForPath( urlPath );
+		return jsonPath.<List<String>> read( "$.statuses[*].state" ).stream().anyMatch( s -> s.equals( "success" ) );
```

Comment:
You should use `success".equals(s)` to avoid NPE and then it can be a method reference.

Reply:
fixed, and some other similar cases